pkg/emitter/emitter.go

```diff
@@ -0,0 +1,313 @@
+package emitter
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
```

Comment:
The general wisdom is that when you need to use reflect package, there probably a nicer and type-safe way to do it.